src/io/dataset.cpp

```diff
@@ -19,8 +19,12 @@
 
 namespace LightGBM {
 
+const float Dataset::kSerializedReferenceVersion = 1.0;
```

Comment:
Maybe use string type like it is used for saved model version?
https://github.com/microsoft/LightGBM/blob/b0137debe6e9cc92b65ec71b0fe8a56ea213c143/src/boosting/gbdt_model_text.cpp#L19

Reply:
Sure.  I made it fixed length, to simplify deserializing. (the example you gave is never deserialized)